### ISSUE TITLE
Enhancement: [no-unused-vars] support reportUsedIgnorePattern

### ISSUE THREAD

**Issue by g1thuser** [typescript-eslint/typescript-eslint#9616]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-unused-vars#options

### Description

The rule should support all options of the base rule like mentioned in the documentation.

### Fail

```TypeScript
https://eslint.org/play#eyJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsic291cmNlVHlwZSI6Im1vZHVsZSJ9fSwidGV4dCI6Ii8qZXNsaW50IG5vLXVudXNlZC12YXJzOiBbXCJlcnJvclwiLCB7IFwicmVwb3J0VXNlZElnbm9yZVBhdHRlcm5cIjogdHJ1ZSwgXCJ2YXJzSWdub3JlUGF0dGVyblwiOiBcIltpSV1nbm9yZWRcIiB9XSovXG5cbnZhciBmaXJzdFZhcklnbm9yZWQgPSAxO1xudmFyIHNlY29uZFZhciA9IDI7XG5jb25zb2xlLmxvZyhmaXJzdFZhcklnbm9yZWQsIHNlY29uZFZhcik7In0=
```

### Pass

```TypeScript
https://eslint.org/play#eyJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsic291cmNlVHlwZSI6Im1vZHVsZSJ9fSwidGV4dCI6Ii8qZXNsaW50IG5vLXVudXNlZC12YXJzOiBbXCJlcnJvclwiLCB7IFwicmVwb3J0VXNlZElnbm9yZVBhdHRlcm5cIjogdHJ1ZSwgXCJ2YXJzSWdub3JlUGF0dGVyblwiOiBcIltpSV1nbm9yZWRcIiB9XSovXG5cbnZhciBmaXJzdFZhciA9IDE7XG52YXIgc2Vjb25kVmFyID0gMjtcbmNvbnNvbGUubG9nKGZpcnN0VmFyLCBzZWNvbmRWYXIpOyJ9
```

**Comment by bradzacher** [comment]
duplicate of #9120
please use the search
it's available on our v8 RC pre-release
